deepchem/metrics/__init__.py

```diff
@@ -19,11 +20,280 @@
+
+
+def recall_score(*args, **kwargs):
+  logger.warning(
+      "recall_score is deprecated. Use sklearn.metrics.recall_score instead. dc.metrics.recall_score will be removed in a future version of DeepChem."
+  )
+  return sklearn.metrics.recall_score(*args, **kwargs)
+
+
+def r2_score(*args, **kwargs):
+  logger.warning(
+      "r2_score is deprecated. Use sklearn.metrics.r2_score instead. dc.metrics.r2_score will be removed in a future version of DeepChem."
+  )
+  return sklearn.metrics.r2_score(*args, **kwargs)
+
+
+def mean_squared_error(*args, **kwargs):
+  logger.warning(
+      "mean_squared_error is deprecated. Use sklearn.metrics.mean_squared_error instead. dc.metrics.mean_squared_error will be removed in a future version of DeepChem."
+  )
+  return sklearn.metrics.mean_squared_error(*args, **kwargs)
+
+
+def mean_absolute_error(*args, **kwargs):
+  logger.warning(
+      "mean_absolute_error is deprecated. Use sklearn.metrics.mean_absolute_error instead. dc.metrics.mean_absolute_error will be removed in a future version of DeepChem."
+  )
+  return sklearn.metrics.mean_absolute_error(*args, **kwargs)
+
+
+def precision_score(*args, **kwargs):
+  logger.warning(
+      "precision_score is deprecated. Use sklearn.metrics.precision_score instead. dc.metrics.precision_score will be removed in a future version of DeepChem."
+  )
+  return sklearn.metrics.precision_score(*args, **kwargs)
+
+
+def precision_recall_curve(*args, **kwargs):
+  logger.warning(
+      "precision_recall_curve is deprecated. Use sklearn.metrics.precision_recall_curve instead. dc.metrics.precision_recall_curve will be removed in a future version of DeepChem."
+  )
+  return sklearn.metrics.precision_recall_curve(*args, **kwargs)
+
+
+def auc(*args, **kwargs):
+  logger.warning(
+      "auc is deprecated. Use sklearn.metrics.auc instead. dc.metrics.auc will be removed in a future version of DeepChem."
+  )
+  return sklearn.metrics.auc(*args, **kwargs)
+
+
+def jaccard_score(*args, **kwargs):
+  logger.warning(
+      "jaccard_score is deprecated. Use sklearn.metrics.jaccard_score instead. dc.metrics.jaccard_score will be removed in a future version of DeepChem."
+  )
+  return sklearn.metrics.jaccard_score(*args, **kwargs)
+
+
+def f1_score(*args, **kwargs):
+  logger.warning(
+      "f1_score is deprecated. Use sklearn.metrics.f1_score instead. dc.metrics.f1_score will be removed in a future version of DeepChem."
+  )
+  return sklearn.metrics.f1_score(*args, **kwargs)
+
+
+def threshold_predictions(y, threshold=0.5):
+  """Threshold predictions from classification model.
+
+  Parameters
+  ----------
+  y: np.ndarray
+    Must have shape `(N, n_classes)` and be class probabilities.
+  threshold: float, optional (Default 0.5)
+    The threshold probability for the positive class. Note that this
+    threshold will only be applied for binary classifiers (where
+    `n_classes==2`). If specified for multiclass problems, will be
+    ignored.
```

Comment:
This creates opportunities for bugs, where users don't realize the argument is being ignored.  Instead how about making the default be `None`, and throwing an exception if they set it to anything else for a multiclass problem?

Reply:
I've swapped over defaults for `n_classes` to `None`